src/peerbook/libp2p_peerbook.erl

```diff
@@ -95,6 +96,24 @@ changed_listener(Pid) ->
 update_nat_type(Pid, NatType) ->
     gen_server:cast(Pid, {update_nat_type, NatType}).
 
+%% @doc Adds an association under the given type to for the swarm this
+%% peerbook is part of. Note that the association _must_ have its
+%% signature be valid for the address of the swarm this peerbook is
+%% part of.
+%%
+%% Associations are gossiped with the peer record for the swarm.
+%%
+%% Note that the given association will replace an existing
+%% association with the given type and address of the association.
+-spec add_association(pid(), AssocType::atom(), Assoc::libp2p_peer:association()) -> ok.
+add_association(Pid, AssocType, Assoc) ->
+    gen_server:cast(Pid, {add_association, AssocType, Assoc}).
+
+-spec lookup_association(pid(), AssocType::atom(), AssocAddress::libp2p_crypto:address())
```

Comment:
a @doc for this function would be nice.

Reply:
uhm.. it's right there.. two lines above line 109

Reply:
no... `lookup_association` has no documentation for it.

Reply:
Fixed